overlord/snapstate/snapstate_update_test.go

```diff
@@ -6861,3 +6861,39 @@ func (s *snapmgrTestSuite) TestUpdatePrerequisiteBackwardsCompat(c *C) {
 		{macaroon: s.user.StoreMacaroon, name: "outdated-consumer", target: filepath.Join(dirs.SnapBlobDir, "outdated-consumer_11.snap")},
 	})
 }
+
+func (s *snapmgrTestSuite) TestUpdateWithHiddenSnapDataDir(c *C) {
```

Comment:
could this test actually check that files are copied over from the previous revision in the hidden dir to the new one? As it stands this test just runs the change and checks that it didn't error out, it would be good to see that we actually do copy files as expected when the change runs

Reply:
The files aren't copied over in this PR, it only adds the setting and things related to using the new directory. The follow-up PR is the one that implements the actual migration and that one has tests for the copying/undoing on refresh and install

Reply:
No what I mean is not the migration of a snap's files from `~/snap/hello-world/1` to `~/.snap/data/hello-world/1`, but rather the copying that takes place when this is enabled for a snap when we refresh from revision 1 to 2, I expect that we will copy from `~/.snap/data/hello-world/1` to `~/.snap/data/hello-world/2` when we refresh, and I don't see a test ensuring that files located in the hidden rev 1 folder will be copied to the hidden rev 2 folder.

For example the test I'm proposing would ensure that we don't accidentally forget to use the option somewhere and on accident we copy files from `~/.snap/data/hello-world/1` to `~/snap/hello-world/2` which would be very bad.

Reply:
Ah sorry, I see what you mean. This test uses the fakeBackend so it can't test the low level part but TestCopyData already covers that. I can add a check to this test to assert that the options are set correctly